### PR TITLE
[RHPAM-4710] EJB_TIMER Ds Auto creation does not recognizes Url XA property

### PR DESCRIPTION
**Pull request by davidesalerno**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[KIECLOUD-XYZ] Subject`, `[RHDM-XYZ] Subject` or `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh [hunk 1]
```diff
@@ -162,6 +162,10 @@ function configure_EJB_Timer_datasource {
 function set_url {
     local prefixedUrl=$(find_env "${1}_URL")
     url=$(find_env "${1}_XA_CONNECTION_PROPERTY_URL" "${prefixedUrl}")
+     # RHPAM-4710 EJB_TIMER Ds Auto creation does not recognizes Url XA property
+    if [ "${url}x" = "x" ]; then
```
Comment (spolti) [inline jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh:166]: is this needed?
the addition on line 200 should be enough to handle it.
if you do this there will *_URL and *_Url set together.

Comment (davidesalerno) [reply jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh:166]: If we don't want that RHPAM_URL is equal to RHPAM__XA_CONNECTION_PROPERTY_Url we can remove this

Comment (spolti) [reply jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh:166]: It is not needed, RHPAM__XA_CONNECTION_PROPERTY_Url  should not be set if RHPAM_URL is there.

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 2]
```diff
@@ -45,6 +45,26 @@ teardown() {
     [ "${RHPAM_XA_CONNECTION_PROPERTY_URL}" = "${RHPAM_URL}" ]
 }
 
+@test "test if the EJB_TIMER datasource has been auto-configured also with RHPAM_XA_CONNECTION_PROPERTY_Url" {
+    local expected_timer_service="EJB_TIMER"
+    local expected_datasources="EJB_TIMER,RHPAM"
+    local expected_url="jdbc:h2:/deployments/data/h2/rhpam;AUTO_SERVER=TRUE"
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:51]: The point of this issue is not exactly what you've changed, but instead, we need to make sure if the user sets the XA property for the main datasource, e.g. RHPAM_ XA_CONNECTION_PROPERTY_Url it is also set for EJBTIMER _XA_CONNECTION_PROPERTY_URL.

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 3]
```diff
@@ -74,6 +94,25 @@ teardown() {
     [ "${EJB_TIMER_XA_CONNECTION_PROPERTY_Url}" = "jdbc:mariadb://myapp-host:3306/rhpam-mariadb?pinGlobalTxToPhysicalConnection=true\&amp;enabledSslProtocolSuites=TLSv1.2" ]
 }
 
+@test "test if the EJB_TIMER datasource has been auto-configured also with RHPAM_XA_CONNECTION_PROPERTY_Url and driver is mariadb" {
+    local expected_timer_service="EJB_TIMER"
+    local expected_datasources="EJB_TIMER,RHPAM"
+    local expected_url="jdbc:mariadb://myapp-host:3306/rhpam-mariadb?enabledSslProtocolSuites=TLSv1.2"
+    export DATASOURCES="RHPAM"
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:101]: please use a full datasource configuration, with driver, and the RHPAM XA Url property.

+ please add a behave test.

File: tests/features/rhpam/rhpam-kieserver.feature [hunk 4]
```diff
@@ -756,7 +756,7 @@ Feature: RHPAM KIE Server configuration tests
     And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value kie_app_mariadb-EJB_TIMER_part on XPath //*[local-name()='database-data-store']/@partition
     And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value 10 on XPath //*[local-name()='xa-pool']/*[local-name()='min-pool-size']
     And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value 10 on XPath //*[local-name()='xa-pool']/*[local-name()='max-pool-size']
-    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value jdbc:mariadb://10.1.1.1:3306/bpms?pinGlobalTxToPhysicalConnection=true&enabledSslProtocolSuites=TLSv1.2 on XPath //*[local-name()='xa-datasource']/*[local-name()='xa-datasource-property'][@name="Url"]
+    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value wip on XPath //*[local-name()='xa-datasource']/*[local-name()='xa-datasource-property'][@name="Url"]
```
Comment (spolti) [inline tests/features/rhpam/rhpam-kieserver.feature:759]: left over?

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats (hunk 2 above)
Comment (spolti) [reply jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:51]: on this bats tests focus on datasources that uses mariadb and postgresql driver.
Same for the other bats tests you've added.

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 5]
```diff
@@ -76,6 +76,25 @@ teardown() {
...
+    export RHPAM_XA_CONNECTION_PROPERTY_Url="${expected_url}"
+    export RHPAM_NONXA="false"
+    configure_EJB_Timer_datasource >&2
+    echo "Expected EJB_TIMER url is ${EJB_TIMER_XA_CONNECTION_PROPERTY_Url}" >&2
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:87]: asdd the RHPAM_DRIVER=mariadb

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 6]
```diff
@@ -179,6 +198,26 @@ teardown() {
...
+    echo "EJBTimer url is ${EJB_TIMER_XA_CONNECTION_PROPERTY_Url} " >&2
+    echo "RHPAM_XA_CONNECTION_PROPERTY_Url is ${RHPAM_XA_CONNECTION_PROPERTY_Url} " >&2
+    [ "${TIMER_SERVICE_DATA_STORE^^}" = "${expected_timer_service}" ]
+    [ "${DATASOURCES}" = "${expected_datasources}" ]
+    [ "${EJB_TIMER_DRIVER}" = "${RHPAM_DRIVER}" ]
+    # we do not expect that this var is set anymore, since we're using URL property directly
+    [ "${EJB_TIMER_XA_CONNECTION_PROPERTY_PinGlobalTxToPhysicalConnection}" = "" ]
+    [ "${EJB_TIMER_XA_CONNECTION_PROPERTY_EnabledSslProtocolSuites}" = "" ]
+    # the URL property must be set
+    [ "${EJB_TIMER_XA_CONNECTION_PROPERTY_Url}" = "jdbc:postgres://host:3306/postgres?ssl=true\&amp\;sslmode=allow" ]
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:218]: Missing the check for the RHPAM xa Url

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats (hunk 2 above)
Comment (davidesalerno) [reply jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats]: I added 2 tests specifically on mysql and postgres, are they enough?

File: tests/features/rhpam/rhpam-kieserver.feature (hunk 4 above)
Comment (davidesalerno) [reply tests/features/rhpam/rhpam-kieserver.feature]: yes, removed

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats (hunk 6 above)
Comment (davidesalerno) [reply jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:219]: Check on RHPAM_URL added

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 7]
```diff
@@ -76,6 +76,25 @@ teardown() {
...
+    export RHPAM_XA_CONNECTION_PROPERTY_Url="${expected_url}"
+    export RHPAM_NONXA="false"
+    configure_EJB_Timer_datasource >&2
+    echo "Expected EJB_TIMER url is ${EJB_TIMER_XA_CONNECTION_PROPERTY_Url}" >&2
+    echo "Expected RHPAM_URL is '${RHPAM_URL}'" >&2
+    echo "Expected RHPAM_XA_CONNECTION_PROPERTY_Url is ${RHPAM_XA_CONNECTION_PROPERTY_Url}" >&2
+    echo "Expected DATASOURCES is ${DATASOURCES}" >&2
+    [ "${TIMER_SERVICE_DATA_STORE^^}" = "${expected_timer_service}" ]
+    [ "${EJB_TIMER_XA_CONNECTION_PROPERTY_Url}" = "jdbc:mariadb://myapp-host:3306/rhpam-mariadb?enabledSslProtocolSuites=TLSv1.2\&amp;pinGlobalTxToPhysicalConnection=true\&amp;enabledSslProtocolSuites=TLSv1.2" ]
+    [ "${RHPAM_XA_CONNECTION_PROPERTY_Url}" = "jdbc:mariadb://myapp-host:3306/rhpam-mariadb?enabledSslProtocolSuites=TLSv1.2" ]
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:94]: This is the expected value for RHPAM_URL, for XA both EJB and RHPAM Url must be the same.

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats [hunk 8]
```diff
@@ -167,7 +186,7 @@ teardown() {
     export RHPAM_JTA="true"
     configure_EJB_Timer_datasource >&2
     echo "EJBTimer url is ${EJB_TIMER_XA_CONNECTION_PROPERTY_Url} " >&2
-    echo "RHPAM url is ${RHPAM_URL} " >&2
+    echo "RHPAM_XA_CONNECTION_PROPERTY_Url is ${RHPAM_XA_CONNECTION_PROPERTY_Url} " >&2
```
Comment (spolti) [inline jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:189]: Any reason to update it?
as it is a nonXA datasource RHPAM_XA_CONNECTION_PROPERTY_Url should not be set.

File: jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats (hunk 6 above)
Comment (spolti) [reply jboss-kie-kieserver/tests/bats/jboss-kie-kieserver.bats:219]: in this case, it is missing to check that RHPAM_URL is empty, it is a XA datasource, RHPAM_URL is for nonXA

**Review by desmax74 — APPROVED** [review]

**Review by desmax74 — APPROVED** [review]

**Pull request closed without merging**